src/tools/crawler/main.rs

```diff
@@ -66,14 +78,24 @@ async fn main() {
     let crawler = Crawler::new().await;
 
     let mut network_metrics = NetworkMetrics::default();
+    let summary_snapshot = Arc::new(Mutex::new(NetworkSummary::default()));
+
+    // Initialize the RPC server if address is specified.
+    let _rpc_handle = if let Some(addr) = args.rpc_addr {
+        let rpc_context = RpcContext::new(Arc::clone(&summary_snapshot));
+        let rpc_handle = initialize_rpc_server(addr, rpc_context).await;
+        Some(rpc_handle)
+    } else {
+        None
+    };
 
     crawler.enable_handshake().await;
     crawler.enable_reading().await;
     crawler.enable_writing().await;
 
     for addr in &args.seed_addrs {
         let crawler_clone = crawler.clone();
-        let addr = addr.parse().unwrap();
+        let addr = *addr;
```

Comment:
nit: let's just dereference directly at line 109

Reply:
seems that this avoids the compiler complaining later on; let's leave it be then :+1: